src/ape_plugins/_cli.py

```diff
@@ -168,7 +168,7 @@ def add(cli_ctx, plugin, version, skip_confirmation, upgrade):
             if result == 0 and is_plugin_installed(plugin):
                 cli_ctx.logger.success(f"Plugin '{plugin}' has been upgraded.")
             else:
-                cli_ctx.logger.error(f"Failed to add '{plugin}'.")
```

Comment:
changed because sometimes the plugin still gets installed / updated even though it was not completely successful.